fix(curriculum): external soundex links in CIP (#46344)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/soundex.md

# --description--

**Soundex Algorithm** deals with the *intentions* of the words. It creates a representation for similar sounding words. 

> It is mainly used in searching for <em>names</em> and <em>addresses</em>, because the person, who inputting the <em>names</em>, can focus on how they sound instead of the correct spelling of the <em>names</em>.

Reviewer: ```suggestion
It is used for searching <em>names</em> and <em>addresses</em>. This means that the person who filled in the <em>name</em>, can focus on how it sounds instead of correcting the spelling of <em>names</em>.
```

For example: 

> If you heard the name `Quenci` for the first time, and misspelled it, you will get **Soundex** code `Q520`. 

Reviewer: ```suggestion
If you are hearing the name `Quenci` for the first time, and misspelled it, you will get **Soundex** code `Q520`. 
```

> When you spell it correctly next time `Quincy`, you will still get the same code `Q520`, which mean you can link multiple name pronunciation into the same <em>person</em> without the need to add every possible spelling. 

Reviewer: ```suggestion
When you spell it correctly next time `Quincy`, you will still get the same code `Q520`, which means you can link multiple name pronunciations into the same <em>person</em> without the need for adding every spelling. 
```
